docs/experimental/ticket-auction.md

```diff
@@ -0,0 +1,258 @@
+
+```sh
+$ tezos-client originate contract nft-wallet transferring 0 from alice \
+       running "$(pwd)/michelson/nft_wallet.tz" \
+        --init "Pair \"tz1LrQB9HrJcUaD9NKEvV65tnaiU8trPXwmt\" (Pair {} (Pair 0 {}))" \
+        --burn-cap 0.305
+
+        { parameter
+                    (or (or (or (pair %auction
+                                   (contract %destination
+                                      (pair (nat %opening_price)
+                                            (pair (nat %set_reserve_price)
+                                                  (pair (timestamp %set_start_time) (pair (int %set_round_time) (ticket %ticket nat))))))
+                                   (pair (nat %opening_price)
+                                         (pair (nat %reserve_price)
+                                               (pair (timestamp %start_time) (pair (int %round_time) (nat %ticket_id))))))
+                                (nat %burn))
+                            (or (map %mint string bytes) (ticket %receive nat)))
+                        (pair %send (contract %destination (ticket nat)) (nat %ticket_id))) ;
+                  storage
+                    (pair (address %admin)
+                          (pair (big_map %tickets nat (ticket nat))
+                                (pair (nat %current_id) (big_map %token_metadata nat (pair nat (map string bytes)))))) ;
+                  code {...}
+
+New contract KT1EAMUQC1yJ2sRPNPpLHVMGCzroYGe1C1ea originated.
+
+```
+#### Bob
+
+Bob originates his nft wallet and sets himself as the admin
+
+```sh
+$ tezos-client originate contract nft-wallet-bob transferring 0 from bob \
+        running "$(pwd)/michelson/nft_wallet.tz" \
+         --init "Pair \"tz1bwfmSYqrhUTAoybGdhWBBefsbuhNdcC2Y\" (Pair {} (Pair 0 {}))" \
+         --burn-cap 1
+
+New contract KT1QQukCBULzFu6samB5FpbLNBXmNzArSpTs originated.
+
+```
+## NFT Auction Contract
+
+A separate NFT auction contract can be used to auction off ones NFTs to prospective buyers. We use a Dutch, descending price auction, in which the asking price of the NFT is decreased until a buyer is found. The NFT auction contract has an entrypoint to `configure` the auction by customizing
+- the `opening_price`
+- the `reserve_price`-- the minimum price the NFT can be reduced to
+- the minimum `start_time` of the auction
+- the `round_time`-- the minimum time the item stays at the same price before its asking price can be decreased.
+- the `ticket`-- the actual NFT that will be auctioned off  
+
+The auction also has entrypoints to `start` the auction, `drop_price` of the NFT after a round has passed, `buy` the NFT by sending the asking price, and an entrypoint for the admin to `cancel` the auction and send the NFT back to their wallet.  
+
+### Auction
+
+Alice originates nft-auction contract
+
+```sh
+$ tezos-client originate contract nft-auction transferring 0 from alice \
+        running "$(pwd)/michelson/nft_auction.tz" \
+         --dry-run \
```

Comment:
it appears that Alice doesn't actually originate anything since this is a dry run